plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadata.java

```diff
@@ -0,0 +1,613 @@
+import io.trino.spi.security.TrinoPrincipal;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeId;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.TypeOperators;
+import io.trino.spi.type.TypeSignature;
+import io.trino.testing.TestingConnectorSession;
+import io.trino.tests.BogusType;
+import org.apache.iceberg.AppendFiles;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DeleteFiles;
+import org.apache.iceberg.ExpireSnapshots;
+import org.apache.iceberg.HistoryEntry;
+import org.apache.iceberg.ManageSnapshots;
+import org.apache.iceberg.ManifestFile;
+import org.apache.iceberg.OverwriteFiles;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.ReplacePartitions;
+import org.apache.iceberg.ReplaceSortOrder;
+import org.apache.iceberg.RewriteFiles;
+import org.apache.iceberg.RewriteManifests;
+import org.apache.iceberg.Rollback;
+import org.apache.iceberg.RowDelta;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.UpdateLocation;
+import org.apache.iceberg.UpdatePartitionSpec;
+import org.apache.iceberg.UpdateProperties;
+import org.apache.iceberg.UpdateSchema;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.plugin.iceberg.TrinoHiveCatalog.DEPENDS_ON_TABLES;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.testcontainers.shaded.com.google.common.base.Preconditions.checkArgument;
+
+public class TestIcebergMetadata
+{
+    private static final CatalogSchemaTableName TESTING_MATERIALIZED_VIEW_NAME =
+            new CatalogSchemaTableName("test_catalog_name", "test_schema_name", "test_table_name");
+
+    private IcebergMetadata testingIcebergMetadata;
+    private TestingTrinoCatalog testingTrinoCatalog;
+
+    @BeforeClass
+    public void setup()
+    {
+        testingTrinoCatalog = new TestingTrinoCatalog();
+        testingIcebergMetadata = new IcebergMetadata(new TestingTypeManager(), jsonCodec(CommitTaskData.class), testingTrinoCatalog);
+    }
+
+    @AfterMethod
+    public void resetTestDependencies()
+    {
+        testingTrinoCatalog.resetState();
+    }
+
+    @Test
+    public void testGetMaterializedViewFreshness()
+    {
+        testingTrinoCatalog.throwWhenOnLoadTable(2);
```

Comment:
Can we make the test simpler?
E.g expect `MaterializedViewNotFoundException` exception on not existing MV and expect result when 
`getMaterializedViewFreshness` is called on mock MV?

What does `throwWhenOnLoadTable` or `returnWhenGetMaterializedView`? It seems they don't even need arguments since they are called just once

Reply:
`throwWhenOnLoadTable` and `returnWhenGetMaterializedView` allows tests to pre-specify what the mock object returns before calling the method under test. I thought they may be helpful since we'll probably add more tests for `IcebergMetadata`. We can remove those options if you'd prefer.

Reply:
yup. I would keep it minimal for now

Reply:
Actually,  I think `returnWhenGetMaterializedView` only needs to be kept to allow `TrinoCatalog` to return different `ConnectorMaterializedViewDefinition` for both test cases.